File: ports/java/src/core/graphics/BcsCanvas.java
package core.graphics;

import java.util.concurrent.locks.ReentrantLock;
import core.kernel.BcsObject;

public class BcsCanvas extends BcsObject {
    private final int width;
    private final int height;
    private final ReentrantLock lock = new ReentrantLock();

    public BcsCanvas(BcsObject parent, int width, int height) {
        super(parent);
        this.width = width;
        this.height = height;
    }

    public int[] getSize() {
        lock.lock();
        try {
            return new int[]{width, height};
        } finally {
            lock.unlock();
        }
    }
}

File: ports/go/src/core/graphics/bcscanvas.go
package graphics

import (
	"sync"
	core "github.com/robertpelloni/btk-go/ports/go/src/core/kernel"
)

type BcsCanvas struct {
	*core.BcsObject
	width  int
	height int
	mu     sync.RWMutex
}

func NewBcsCanvas(parent *core.BcsObject, width, height int) *BcsCanvas {
	return &BcsCanvas{
		BcsObject: core.NewBcsObject(parent),
		width:     width,
		height:    height,
	}
}

func (c *BcsCanvas) Size() (int, int) {
	c.mu.RLock()
	defer c.mu.RUnlock()
	return c.width, c.height
}
